### PR TITLE
feat(fixer): add Prowler Fixer feature!

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add Prowler Fixer feature to fix checks when Prowler finds misconfigurations. At the moment, it is only included for check `ec2_ebs_default_encryption`.

`prowler aws -c ec2_ebs_default_encryption --fixer`
image

`prowler aws --list-fixer`
image

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/check/check.py [hunk 1]
```diff
@@ -419,12 +419,33 @@ def run_check(check: Check, output_options) -> list:
         return findings
 
 
+def run_fixer(check_findings, check_name, c):
```
Comment (jfagoagas) [inline prowler/lib/check/check.py]: What is `c`? Please use a more readable word, is free of cost!

File: prowler/providers/aws/services/ec2/ec2_service.py [hunk 2]
```diff
@@ -410,6 +410,18 @@ def __get_ebs_encryption_settings__(self, regional_client):
                 f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __enable_ebs_encryption_by_default__(self, region):
```
Comment (jfagoagas) [inline prowler/providers/aws/services/ec2/ec2_service.py]: Add a test for this in the service. Thanks!

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/check/check.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/lib/check/check.py]: Done!

File: prowler/providers/aws/services/ec2/ec2_service.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/ec2/ec2_service.py]: Done!

File: prowler/lib/check/check.py [hunk 3]
```diff
@@ -419,9 +427,16 @@ def run_check(check: Check, output_options) -> list:
         return findings
 
 
-def run_fixer(check_findings, check_name, c):
+def run_fixer(check_findings: list, check_name: str, check_class: Check):
+    """
+    Run the fixer for the check if it exists and there are any FAIL findings
+    Args:
+        check_findings (list): list of findings
+        check_name (str): check name
```
Comment (jfagoagas) [inline prowler/lib/check/check.py]: I think you don't need the check name here, you can use `check.CheckID`.

File: tests/providers/aws/services/ec2/ec2_service_test.py [hunk 4]
```diff
@@ -544,3 +544,19 @@ def test__describe_volumes__(self):
         assert ec2.volumes[0].tags == [
             {"Key": "test", "Value": "test"},
         ]
+
+    # Test EC2 EBS Enabling Encryption by Default
```
Comment (jfagoagas) [inline tests/providers/aws/services/ec2/ec2_service_test.py]: lovely!

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/banner.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ def print_banner(args):
 """
     print(banner)
 
-    if args.verbose:
+    if args.verbose or getattr(args, "fix", None):
```
Comment (jfagoagas) [inline prowler/lib/banner.py:18]: If this needs to be done force the `verbose` to be `True` if the `fix` is set, I think is more than enough to have one argument to control the outputs.

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**